packages/transactions/src/builders.ts

```diff
@@ -139,13 +139,172 @@ export async function estimateTransfer(
+export type DeserializationRejection = {
+  error: string;
+  reason: TxRejectedReason.Deserialization;
+  reason_data: {
+    message: string;
+  }
+  txid: string;
+};
+
+export type SignatureValidationRejection = {
+  error: string;
+  reason: TxRejectedReason.SignatureValidation;
+  reason_data: {
+    message: string;
+  }
+  txid: string;
+};
+
+export type BadNonceRejection = {
+  error: string;
+  reason: TxRejectedReason.BadNonce;
+  reason_data: {
+    expected: number;
+    actual: number;
+    is_origin: boolean;
+    principal: boolean;
+  }
+  txid: string;
+};
+
+export type FeeTooLowRejection = {
+  error: string;
+  reason: TxRejectedReason.FeeTooLow;
+  reason_data: {
+    expected: number;
+    actual: number;
+  }
+  txid: string;
+};
+
+export type NotEnoughFundsRejection = {
+  error: string;
+  reason: TxRejectedReason.NotEnoughFunds;
+  reason_data: {
+    expected: string;
+    actual: string;
+  }
+  txid: string;
+};
+
+export type NoSuchContractRejection = {
+  error: string;
+  reason: TxRejectedReason.NoSuchContract;
+  txid: string;
+};
+
+export type NoSuchPublicFunctionRejection = {
+  error: string;
+  reason: TxRejectedReason.NoSuchPublicFunction;
+  txid: string;
+};
+
+export type BadFunctionArgumentRejection = {
+  error: string;
+  reason: TxRejectedReason.BadFunctionArgument;
+  reason_data: {
+    message: string;
+  }
+  txid: string;
+};
+
+export type ContractAlreadyExistsRejection = {
+  error: string;
+  reason: TxRejectedReason.ContractAlreadyExists;
+  reason_data: {
+    contract_identifier: string;
+  }
+  txid: string;
+};
+
+export type PoisonMicroblocksDoNotConflictRejection = {
+  error: string;
+  reason: TxRejectedReason.PoisonMicroblocksDoNotConflict;
+  txid: string;
+};
+
+export type PoisonMicroblockHasUnknownPubKeyHashRejection = {
+  error: string;
+  reason: TxRejectedReason.PoisonMicroblockHasUnknownPubKeyHash;
+  txid: string;
+};
+
+export type PoisonMicroblockIsInvalidRejection = {
+  error: string;
+  reason: TxRejectedReason.PoisonMicroblockIsInvalid;
+  txid: string;
+};
+
+export type BadAddressVersionByteRejection = {
+  error: string;
+  reason: TxRejectedReason.BadAddressVersionByte;
+  txid: string;
+};
+
+export type NoCoinbaseViaMempoolRejection = {
+  error: string;
+  reason: TxRejectedReason.NoCoinbaseViaMempool;
+  txid: string;
+};
+
+export type ServerFailureNoSuchChainTipRejection = {
+  error: string;
+  reason: TxRejectedReason.ServerFailureNoSuchChainTip;
   txid: string;
 };
+
+export type ServerFailureDatabaseRejection = {
+  error: string;
+  reason: TxRejectedReason.ServerFailureDatabase;
+  reason_data: {
+    message: string;
+  }
+  txid: string;
+};
+
+export type ServerFailureOtherRejection = {
+  error: string;
+  reason: TxRejectedReason.ServerFailureOther;
+  reason_data: {
+    message: string;
+  }
+  txid: string;
+};
+
+export type TxBroadcastResultOk = {
+  txid: string;
+};
+
+export type TxBroadcastResultRejected =   SerializationRejection |
```

Comment:
@kyranjamie any idea why this is passing lint checks? Locally, vscode inline shows it as a prettier error.

Reply:
It's an error. It's because it's running from a fork, and the jobs are disabled there, it seems. Or just aren't running at all.

Reply:
@ahsan-javaid can you make sure the lint/formatting is fixed? In the future, you should open PRs from a branch off this repo (you should have maintainer permission for that), and github actions will catch these errors.

Reply:
@zone117x 
- Alright will make PR's from branch within the repo in future pull requests
- Thanks for the above example. It helped a lot to understand the scenario. I have added the changes to incorporate the error scenario. Now the error can be checked easily with if condition. Great idea :)
```
// This works now.
 const reply: TxBroadcastResult = await broadcastTransaction(
      transaction, network
  );
  if (reply.error) {
    console.log(reply.reason);
  } else {
    console.log(reply.txid);
  }
``` 
- lint errors are also removed and `npm run lint` passes successfully